src/amber/router/routed_result.cr

```diff
@@ -1,34 +1,32 @@
 module Amber::Router
   class RoutedResult(T)
+    include Comparable(RoutedResult)
+
     getter params = {} of String => String
 
     def initialize(@terminal_segment : TerminalSegment(T)?)
     end
 
-    delegate :[], :[]=, to: @params
+    delegate :[]?, :[], :[]=, to: @params
 
     def terminal_segment
       @terminal_segment.not_nil!
     end
 
     def path
-      if @terminal_segment
+      if found?
         terminal_segment.full_path
       else
         raise "Cannot provide route path when no route was found. Ask first with #found?"
       end
     end
 
     def found?
-      !@terminal_segment.nil?
+      !!@terminal_segment
```

Comment:
small bike shedding but consider avoiding `!!`.  I find `.nil?` preferable.

Reply:
`!!` is pretty known idiom, personally I find it more readable than `!var.nil?` - double negation split into prefix and postfix making my mind needlessly jump instead of having it directly upfront.

Reply:
I know.  I see it all the time.  I'm not a fan and consider it a hack from the JS world.  I'm just bike shedding.

Reply:
@drujensen I agree with you.

Reply:
Ok, I'll revert it back.